ice40/arch.cc

```diff
@@ -642,6 +643,12 @@ std::vector<GroupId> Arch::getGroupGroups(GroupId group) const
 
 bool Arch::place()
 {
+#ifdef USE_RUST
```

Comment:
Suggestion to avoid this `ifdef`: add a Python binding for this function conditional on `USE_RUST` so it can be called as desired

Reply:
> (said example is intended to be removed before merge, but exists as a demonstration that the integration works)

That's meant to cover this; I never intended that to get merged, but as a "look, you can compile it yourself and see that it works" type example.

Reply:
Yep, I saw, it was just my suggestion if you wanted to keep the demo accessible but remove this code

Reply:
Okay, pybind11 confuses me, so I'm going to have to ask for your help with writing said binding.

Reply:
I think just adding the necessary include at the top of the file and then adding

```
m.def("example_printnets", example_printnets);
```
after say (and with the necessary ifdef guard)
https://github.com/YosysHQ/nextpnr/blob/56587859d3596c4a42d5f28ae51898ee09cea2c1/common/kernel/pybindings.cc#L251

might work - although it might be you will need to make a function that calls example_printnets taking `Context&` instead of `Context*`

Reply:
ping?

Reply:
with your patch, this now works :p